Compute update folder structure

grizzlytheodore opened the pull request:
Swagger folders restructure for to follow [Swagger team’s guideline](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fblob%2FUpdateDocumentation%2FREADME.md%23folder-structure-for-service-group&data=05%7C01%7CTheodore.Chang%40microsoft.com%7C97a4c52ffa1d46f5dcb108da54926f00%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637915285153599585%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=JliprLBBS0x2AbqCEGeS8AioizYXZzVAKzPGH1JiDsA%3D&reserved=0) and for improvement in developer experience.

What is happening in this PR is we are moving all the versions in the Microsoft.Compute to "previous-versions" folder so that going forward, we can distinguish different RP's folders from the Microsoft.Compute folder THEN go into different versions. 

We have few releases, that are adopting this already, 2022-08-01 CRP release https://github.com/Azure/azure-rest-api-specs/tree/feature/cplat-2022-08-01
 and 2022-03-03 galleryRP release https://github.com/Azure/azure-rest-api-specs/tree/feature/gallery-2022-03-03-2nd 
which are both being worked on, but they both have 3000+ changes because of moving the files around. So we wanted to do it in this separate PR to keep  new version changes, and restructing changes isolated. 

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

grizzlytheodore committed:
update folder structure

grizzlytheodore committed:
update folder structure

grizzlytheodore committed:
small errors and CI check

grizzlytheodore committed:
small errors and CI check

grizzlytheodore commented:
CI failures seems to be false alarm/stale warnings from versions long ago, by changing the path of all versions

grizzlytheodore commented:
I am able to generate SDK and it shows no change

grizzlytheodore commented:
please review @ArcturusZhang

Commit:
Merge branch 'main' into feature/cplat-swaggerRestructure

Commit:
Merge branch 'main' into feature/cplat-swaggerRestructure

deyaaeldeen committed:
[Language Text] Update swagger titles (#19835)

* [Language Text] Update swagger titles

* edits

* address feedback

FumingZhang committed:
Review request for Microsoft.ContainerService to add version 2022-06-01 (#19848)

* Adds base for updating Microsoft.ContainerService from version stable/2022-04-01 to version 2022-06-01

* Updates readme

* Updates API version in new specs and examples

* update readmes (#19421)

* Add key management service profile of a managed cluster for version 2022-06-01 (#19529)

* Add NetworkPlugin none option to 2022-06-01 (#19510)

* Add NetworkPlugin none option to 2022-06-01

* improve description for none value

* Fix violated rule R4041 for 2022-06-01 managedCluster swagger (#19581)

* remove useless directive

* fix R4041

* GA AKS support for Dedicated Host Group (#19547)

* GA AKS support for Dedicated Host Group

* fit some format minor issue

* add newline in end of files

* Defender updates (#19665)

* Defender updates

* sample

* remove old description

Co-authored-by: Bin Xia <binxi@microsoft.com>
Co-authored-by: Matt Stam <mattstam@live.com>
Co-authored-by: Jianping Zeng <zjpjack@users.noreply.github.com>
Co-authored-by: Or Parnes <orparnes@microsoft.com>

abatishchev committed:
Update resources.json (#19861)

PatrickTseng committed:
Pattts/predictive autoscale 20221001 (#19765)

* Rev Autoscale API from 2021-05-01-preview to 2022-10-01.

* Fix minor swagger issue. Default should be 'false'

* Fix swagger validation error

* Defining a new package 2022-10

* AutoscaleSettingResource now reference allOf 'resource' which is the actual autoscaleSetting resource definition

* Fix require property issue

* Run prettier against autoscale_API.json

gigigoo0 committed:
[Maps - Render & Spatial Services] add v2022-08-01 (#19520)

* copy old swagger to new folders

* set api version to 2022-08-01

* apply api changes

Co-authored-by: Gigi Grajo <gigigrajo@microsoft.com>

Wzb123456789 committed:
update appconfiguration (#19330)

* update appconfiguration

* Update readme.python.md

* Update readme.python.md

* Update readme.python.md

* Update readme.python.md

* Update readme.python.md

* Update readme.python.md

* Update readme.python.md

AbhishekKrishna123 committed:
[Hub Generated] Publish private branch 'main' (#19852)

* Add StorageMover specification for PubliC Preview

* SpellCheck fix

* Update custom-words

* Update specification/storagemover/resource-manager/readme.md

Co-authored-by: Abhishek Krishna <abkrish@microsoft.com>
Co-authored-by: Dapeng Zhang <dapzhang@microsoft.com>

divyansh3131 committed:
Merging Dynatrace swagger in stable folder (#19862)

* Committing base swagger version

* Removing preview tag from stable folder api version

* Changing version name in all the files

* Updating readme.go and readme.md files

t-bzhan committed:
[CDN] Fix customDomains property type in Endpoint (#19788)

* [CDN] Fix customDomains property type in Endpoint

* Fix linter error

Co-authored-by: Bo Zhang <bzhan@microsoft.com>

cohen-uriel-ms committed:
Adding Microsoft.Sql AdvancedThreatProtectionSettings APIs for MI on v5 tag for 2022-02-01-preview (#19866)

grizzlytheodore committed:
update folders

grizzlytheodore committed:
update folders

susolank committed:
[Hub Generated] Review request for Microsoft.KeyVault to add version stable/7.3 (#19844)

* Update description of exportable attribute and release_policy.data

* Remove submodule azure-reset-api-specs

* Fix description for exportable

Co-authored-by: Sunny Solanki <Sunny.Solanki@microsoft.com>

AbhishekKrishna123 committed:
Add latest StorageMover updates (#19876)

* Add StorageMover specification for PubliC Preview

* SpellCheck fix

* Update custom-words

* Update specification/storagemover/resource-manager/readme.md

* Add latest StorageMover updates

Co-authored-by: Abhishek Krishna <abkrish@microsoft.com>
Co-authored-by: Dapeng Zhang <dapzhang@microsoft.com>

Wzb123456789 committed:
Update readme.python.md (#19899)

* Update readme.python.md

* Update readme.md

bexxx committed:
Moving files from azure-rest-spec-pr to azure-rest-spec repo after api review (#19878)

* Moving files from azure-rest-spec-pr to azure-rest-spec repo after api review

* Add known words

arpit-gagneja committed:
Removed static IP allocation only from examples (#19858)

* Removed static IP allocation

* Removed static only from examples

Co-authored-by: Arpit Gagneja <argagnej@microsoft.com>

Wzb123456789 committed:
Update securityinsights readme.python (#19903)

* update securityinsights readme.python

* Update readme.python.md

Co-authored-by: Zhenbiao Wei (WICRESOFT NORTH AMERICA LTD) <v-zhenbwei@microsoft.com>

yanjungao718 committed:
fix lint errors in Synapse trigger.json (#19660)

grizzlytheodore committed:
put json files into RP folders

grizzlytheodore committed:
put json files into RP folders

grizzlytheodore committed:
update readme

grizzlytheodore committed:
update readme

grizzlytheodore committed:
Merge branch 'feature/cplat-swaggerRestructure' of https://github.com/Azure/azure-rest-api-specs into feature/cplat-swaggerRestructure

Commit:
Add CONTRIBUTING.md (#19257)

* Add CONTRIBUTING.md

* Apply suggestions from PR review

Co-authored-by: Heath Stewart <heaths@outlook.com>

* Apply suggestions from PR review

Co-authored-by: Weidong Xu <weidxu@microsoft.com>

* Regen toc for CONTRIBUTING.md

* Address PR review comments

* Address PR review comments

Co-authored-by: Heath Stewart <heaths@outlook.com>
Co-authored-by: Weidong Xu <weidxu@microsoft.com>

Alancere committed:
fix devcenter readme.go.md config (#19906)

* fix readme.go.md config

* rename to SkuInfo

ArcturusZhang reviewed (commented):
The cloudService RP has a casing issue, for instance here: specification/compute/resource-manager/Microsoft.Compute/CloudserviceRP/preview/2020-10-01-preview/cloudService.json

j-zhong-ms committed:
add annotation for labservices (#19884)

tfitzmac committed:
add to description of OS state (#19764)

neeerajaakula committed:
[Hub Generated] Review request for Microsoft.KeyVault to add version preview/2021-06-01-preview (#19767)

* Updated the managed hsm resource manager spec to include two additional properties for private endpoint connection item

* Update managed hsm private endpoint connection item in mhsm spec for latest api version

* Address LRO_RESPONSE_HEADER violation for managed hsm

Long running operations that are annotated with x-ms-long-running-operation:true
must return location header or azure-AsyncOperation in response.
Added the missing location header for managed hsm update command as well as to the corresponding
examples.

darkoa-msft committed:
[Microsoft.DeviceUpdate] Adding first stable API version (#19846)

* Copied the most recent preview version into the first stable version

* Updated api version in new stable version, added the tag to readme.md

* Fixed typo

* Fixed the wrong path in readme.md

* Reverted VS Code automatic breaking change

roy-work committed:
Fix broken link (#19688)

amitelran committed:
add aadObjectId property to kustoPool (#19856)

Co-authored-by: Amit Elran <amelran@microsoft.com>

Wzb123456789 committed:
Updata securityinsights readme.python (#19917)

* update securityinsights readme.python

* update readme.python

Co-authored-by: Zhenbiao Wei (WICRESOFT NORTH AMERICA LTD) <v-zhenbwei@microsoft.com>

kazrael2119 committed:
modify readme.typescripy.md for compute package (#19607)

Co-authored-by: ZiWei Chen (WICRESOFT NORTH AMERICA LTD) <v-ziweichen@microsoft.com>

gubalasu committed:
Dev gubalasu frontdoor microsoft.network 2021 06 01 (#19578)

* Adds base for updating Microsoft.Network from version stable/2020-05-01 to version 2021-06-01

* Updates readme

* Updates API version in new specs and examples

* update waf configs and examples from 2020-11-01

* Update api version in new specs and examples

* Migration api update

* Update readme

* Fix readme

* update readme

* Fix readme again

* Fix readme

* Fix readme

* update waf policy provisioning state

Wzb123456789 committed:
Update securityinsights readme.python (#19923)

* update securityinsights readme.python

* update readme.python

* last modify

* Update readme.python.md

Co-authored-by: Zhenbiao Wei (WICRESOFT NORTH AMERICA LTD) <v-zhenbwei@microsoft.com>

grizzlytheodore committed:
sync with changes made to master

grizzlytheodore committed:
resolve conflict

grizzlytheodore committed:
resolve capitalization and number misplacement accident

ArcturusZhang commented:
The model validation and avocado are all reported in old api-versions.

ArcturusZhang commented:
Also suppress the breaking change alarm - this PR only changes the structure of files, no file content changes.

ArcturusZhang commented:
/azp run
